python/tank/platform/application.py

```diff
@@ -207,6 +209,17 @@ def log_exception(self, msg):
     ##########################################################################################
     # internal API
 
+    def event(self, event):
```

Comment:
I think this method should be protected, e.g. with single underscore, to indicate that you should not call it from external code (engine base class always calls it) but applications can subclass it if they want to listen to incoming events.

```
def _engine_event(event):
    # event object is always a EngineEvent
```

If we were to follow the QT paradigm - which yes, is a little C++, so maybe not relevant, we could add 

```
def _file_open_event(event):
    # event object is always a FileOpenEvent
```

I can see this being quite convenient for a TD to work with and makes the resulting code quite readable. May also give us a good anchor to hang the sphinx docs on. What do you think?